src/main/scala/com/fulcrumgenomics/umi/DuplexConsensusCaller.scala

```diff
@@ -203,7 +203,10 @@ class DuplexConsensusCaller(override val readNamePrefix: String,
         // Convert to SamRecords and return
         (duplexR1, duplexR2) match {
           case (Some(r1), Some(r2)) =>
-            Seq(createSamRecord(r1, FirstOfPair), createSamRecord(r2, SecondOfPair))
+            Seq(
+              createSamRecord(r1, FirstOfPair, toUmiBases(filteredAbR1s ++ filteredBaR2s)),
```

Comment:
Looking at which `++` gets called, these calls to `filteredAbR1s ++ filteredBaRs2` do appear to construct a new collection each time and add all the elements.  Might be nice to just store the collections before lines 200-201 and reuse them here.